kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceChannelProvider.java

```diff
@@ -0,0 +1,255 @@
+          ptvElectronicChannel.add(electronicChannel);
+        }
+      }
+    }
+   
+    return translateChannels(ptvElectronicChannel);
+  }
+  
+  private List<ServiceElectronicChannel> translateChannels(List<VmOpenApiElectronicChannel> ptvChannels) {
+    if (ptvChannels == null) {
+      return Collections.emptyList();
+    }
+    
+    List<ServiceElectronicChannel> result = new ArrayList<>();
+    for (VmOpenApiElectronicChannel ptvChannel : ptvChannels) {
+      result.add(translateChannel(ptvChannel));
+    }
+    
+    return result;
+  }  
+  
+  private ServiceElectronicChannel translateChannel(VmOpenApiElectronicChannel ptvChannel) {
+    ServiceChannelId ptvId = new ServiceChannelId(PtvConsts.IDENTIFIFER_NAME, ptvChannel.getId());
+    
+    ServiceChannelId kuntaApiId = idController.translateServiceChannelId(ptvId,  KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiId == null) {
+      Identifier identifier = identifierController.createIdentifier(IdType.SERVICE_CHANNEL, ptvId.getSource(), ptvId.getId());
+      logger.info(String.format("Discovered new electric service channel %s", ptvId.toString()));
+      kuntaApiId = new ServiceChannelId(KuntaApiConsts.IDENTIFIER_NAME, identifier.getKuntaApiId());
+    }
+    
+    ServiceElectronicChannel result = new ServiceElectronicChannel();
+    result.setAttachments(translateAttachments(ptvChannel.getAttachments()));
+    result.setUrls(translateLanguageItems(ptvChannel.getUrls()));
+    result.setRequiresAuthentication(ptvChannel.getRequiresAuthentication());
+    result.setSignatureQuantity(ptvChannel.getSignatureQuantity());
+    result.setSupportContacts(translateSupportContacts(ptvChannel.getSupportContacts()));
+    result.setServiceHours(translateServiceHours(ptvChannel.getServiceHours()));
+    result.setWebPages(translateWebPages(ptvChannel.getWebPages()));
+    result.setNames(translateLocalizedItems("Name", ptvChannel.getServiceChannelNames()));
+    result.setShortDescriptions(translateLocalizedItems("ShortDescription", ptvChannel.getServiceChannelNames()));
+    result.setDescriptions(translateLocalizedItems("Description", ptvChannel.getServiceChannelNames()));
+    result.setId(kuntaApiId.getId());
+    
+    return result;
+  }
+
+  private List<ServiceChannelWebPage> translateWebPages(List<VmOpenApiWebPage> ptvWebPages) {
+    if (ptvWebPages == null) {
+      return Collections.emptyList();
+    }
+    
+    List<ServiceChannelWebPage> result = new ArrayList<>();
+    
+    for (VmOpenApiWebPage ptvWebPage : ptvWebPages) {
+      ServiceChannelWebPage webPage = new ServiceChannelWebPage();
+      webPage.setLanguage(ptvWebPage.getLanguage());
+      webPage.setType(ptvWebPage.getType());
+      webPage.setUrl(ptvWebPage.getUrl());
+      webPage.setValue(ptvWebPage.getValue());
+      result.add(webPage);
+    }
+    
+    return result;
+  }
+
+  private List<ServiceChannelServiceHour> translateServiceHours(List<VmOpenApiServiceHour> ptvServiceHours) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateServiceHours" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)